packages/webdriver/src/bidi/core.ts

```diff
@@ -54,18 +55,52 @@ export class BidiCore {
             return
         }
 
-        this.#waitForConnected = new Promise<boolean>((resolve) => {
-            this.#ws.on('open', () => {
-                log.info('Connected session to Bidi protocol')
+        log.info(`Connecting to webSocketUrl ${this.#webSocketUrl}`)
+        const parsedUrl = new URL(this.#webSocketUrl)
+        // https://github.com/webdriverio/webdriverio/issues/14039
+        const candidateUrls: string[] = [this.#webSocketUrl]
+        if (!isIP(parsedUrl.hostname)) {
+            const [ips4, ips6] = await Promise.all([
+                dns.promises.resolve4(parsedUrl.hostname),
+                dns.promises.resolve6(parsedUrl.hostname),
+            ])
+            const candidateIps = [...ips4, ...ips6]
+            candidateUrls.push(
+                ...candidateIps.map((ip) => this.#webSocketUrl.replace(parsedUrl.hostname, ip))
+            )
+        }
+
+        this.#waitForConnected = (async () => {
+            const wsConnectPromises: Promise<WebSocket | null>[] = []
+            for (const candidateUrl of candidateUrls) {
+                const connectPromise = new Promise<WebSocket | null>((resolve) => {
+                    const ws = new environment.value.Socket(candidateUrl, this.#clientOptions) as unknown as WebSocket
```

Comment:
Could you try getting the correct type without type casting?

Reply:
I am not sure this is possible. It was already done like that before, I did not change the implementation.
Do you have any proposals on how this particular line could be improved?

Reply:
This cast already exist before this patch, happy to improve it but can remain outside of the PR